### DISCUSSION TITLE
Upload files with other input in the same form

### DISCUSSION THREAD

**Discussion by paroxyste-0** [miguelgrinberg/microdot#248] [Q&A]
Hello, I would like to send a form with input text, select and textarea at the same time as 2 input files.

I have increased the size of Request to avoid error 413.

My problem is that if I put application/x-www-form-urlencoded I receive the information from my inputs and I have the name of the uploaded images, but I can't retrieve their content with request.form.

If I use application/octet-stream I get the same information as before with request.form or await request.stream.read()

If I use multipart/form-data form and body are empty and using await request.stram.read() takes an infinitely long time to get any response.

What should I do? On flask, you can use request.files, but on microdot, the examples indicate that you should use stream, but I don't understand how to retrieve the contents of my images in addition to my form.

Thanks

**Comment by miguelgrinberg** [comment]
The correct thing to do here is to use `multipart/form-data`, but unfortunately this isn't implemented in Microdot yet so you would need to parse and decode the body yourself. I have it on the [roadmap](https://github.com/miguelgrinberg/microdot?tab=readme-ov-file#roadmap), so I'm aware of the problem and intend to work on it.

**Comment by paroxyste-0** [comment]
Thank you for your reply and the information. Do you think you can implement it during the rest of the year?

Sincerely

**Reply by miguelgrinberg** [reply]
I can't promise that, sorry.
